rfcs/0001-tokens-naming-convention.md

```diff
@@ -0,0 +1,495 @@
+2. If I have to add `fontFamily` as a token in the theme I don't know what structure shall I follow? `theme.fontFamily.name` or `theme.font.family.name` or `theme.typography.fontFamily`. All of them look correct at a first glance but the decision making becomes difficult.
+3. I've a Button component and I want to set the text color of a Button when it's in hovered state. Now I don't know whether `Button.color.text.hover` is the right way or `Button.hover.color.text` is the right way.
+
+It's quite evident that we need a system that's flexible and also specific. 
+
+We need a well descriptive tokenized language that incorporates all these concerns and gives us a structure in a hierarchical manner. The language should be elaborative enough that accessing the tokens almost becomes self explanatory. This means we need to start defining some kind of groups.
+
+## Proposed Structure
+The proposed grouping that will fit our use case would look something like `object.base.modifier`.
+<br/>
+<br/>
+A picture worth thousand words 😄
+
+![image](./images/token-naming-overview.png)
+
+
+
+Let's look at the breakdown for each of the group:
+
+### Object
+An object level classifies tokens specific to a component, element nested within a component, or a component group or even a theme to differentiate whether it's a theme level global token or component level local token. `Object` level will become the first level of classification in our new token structure.
+
+For example if we don't use this specifier then we might end up creating a token like `color.background` for let's say a `Button` component. But this leads to an ambiguity since we we don't have any context attached to the color and we don't know what color we are referring to. 
+
+Rather if we start the token naming with a `Base` which means prefixing the component name then things might become more clear and contextual. For eg: `Button.color.background`
+
+But now if we want to refer a component within a component for example we need to refer to the `color` of an `icon` component inside a `Button` component then how shall we do that? An object can also split further and can have sub-component nesting to it as well. Eg: `Button.leftIcon.color.background` 
+
+Similarly, if we want to refer to a token value within a theme like `font.family` we can't do it alone since it doesn't has any context attached to it and we don't even know if what are we referring to? a theme token or a component token?
+
+But as soon as we attach the context our problem will be solved. For eg: `theme.font.family` tells us that we are trying to refer to font family token inside our theme.
+
+#### TL;DR
+**Object** refers to
+  1. **component** - `Button`, `theme` etc.
+  2. **sub-component** - one or more component within a component - `left-icon`, `right-icon` etc.
```

Comment:
Are we calling this element or sub-component?

Reply:
Element is a generic term. A component is an element and a sub-component is also an element. But you never will use these keywords it's just to set the meaning. Hence it says `one or more component within a component` along with the example